### DISCUSSION TITLE
Can you run tests from separate repo at build time

### DISCUSSION THREAD

**Discussion by HuckleberryCarignan** [community/community#38295] [Actions]
My advanced apologies - I am a complete N00b when it comes to GitHub Actions.   My question is:
    If you are building from repo "X" can you run test in repo "Y"?
This is in regards to some end-to-end tests we have in a separate repo.

Thanks,
Huck

**Comment by david-engelmann** [comment]
if repo "X" is a dependency of repo "Y", you could install and test repo "X" from the repo "Y". I do this with internal dependencies. I use a typical testing flow such as pytest. In these cases repo "Y" is the best place to run integration test as well.

**Comment by airtower-luna** [comment]
If you mean that you want to use multiple repositories in your workflow, see the [`actions/checkout` documentation on checking out multiple repositories](https://github.com/actions/checkout#checkout-multiple-repos-side-by-side).

Whether it makes sense to do it that way depends a lot on how the repositories are connected, as @david-engelmann pointed out.

**Reply by david-engelmann** [reply]
Totally, If they are completely separate, It probably makes the most sense to test them separately. Results are clearer and resolutions are simpler when what you are testing is smaller

**Reply by HuckleberryCarignan** [reply]
Details - I'm building some playwright end-to-end tests in repo "Y".    When I build either repo "X" or "Z", I want to kick off the tests defined in repo "Y".    It sound like I can do this.  Do I have that correct?

**Reply by david-engelmann** [reply]
Ok that makes sense to me. So basically, "X" and "Z" are dependencies of "Y". You'll need to create a workflow in both "X" and "Z" to modify repo "Y" when "X" or "Z" are merged or what not.

Here's the general flow
* Trigger Action in "X" or "Z"
* Action creates an Issue in "Y"
* Issue Created in "Y" triggers the running of the full test suite for "X", "Y" and "Z" within "Y"

**Reply by airtower-luna** [reply]
I'd use `workflow_dispatch` rather than creating an issue (or commit), but otherwise yes. X and Z would have to send the event to Y on any change that should create a test run. :slightly_smiling_face: 

See: [Manually running a workflow](https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow)

**Comment by HuckleberryCarignan** [comment]
Where can I find documentation describing this ability (so I can try it)?

**Reply by david-engelmann** [reply]
[This forum post might be able to add some clarity](https://www.reddit.com/r/github/comments/icvqbr/is_there_a_way_to_configure_a_action_to_listen_to/)

**Comment by Lina-hdez** [comment]
> Ok that makes sense to me. So basically, "X" and "Z" are dependencies of "Y". You'll need to create a workflow in both "X" and "Z" to modify repo "Y" when "X" or "Z" are merged or what not.
> 
> Here's the general flow
> 
> * Trigger Action in "X" or "Z"
> * Action creates an Issue in "Y"
> * Issue Created in "Y" triggers the running of the full test suite for "X", "Y" and "Z" within "Y"

so, if tests in repo "Y" fail, does that prevent the action in repos "X" and "Y" or it does not afect... it will work just as a log or something?

**Reply by airtower-luna** [reply]
As such, a workflow in repo Y just produces a workflow report there.

If you want anything else you'd have to set it up yourself, e.g. API calls to add a status to the commit that triggered the build, or sending mails about the failure. You'd probably need credentials, if you do make sure the code from the other repositories can't extract those, see: https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
